docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,189 @@ weight: 700
+| Another field of the user information from the UserInfo endpoint.                  | Set `name_attribute_path` configuration option. |
+
+### Configure email address
+
+Grafana can resolve the user's email address from the OAuth2 ID token, the user information retrieved from the OAuth2 UserInfo endpoint, or the OAuth2 `/emails` endpoint.
+Grafana looks at these sources in the order listed until an email address is found.
+If no email is found, then the email address of the user is set to an empty string.
+
+Refer to the following table for information on what to configure based on how the Oauth2 provider returns a user's email address:
+
+| Source of email address                                                                                                                                                 | Required configuration                                                                                             |
+| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `email` field of the OAuth2 ID token.                                                                                                                                   | N/A                                                                                                                |
+| `attributes` map of the OAuth2 ID token.                                                                                                                                | Set `email_attribute_name` configuration option. By default, Grafana searches for email under `email:primary` key. |
+| `upn` field of the OAuth2 ID token.                                                                                                                                     | N/A                                                                                                                |
+| `email` field of the user information from the UserInfo endpoint.                                                                                                       | N/A                                                                                                                |
+| Another field of the user information from the UserInfo endpoint.                                                                                                       | Set `email_attribute_path` configuration option.                                                                   |
+| Email address marked as primary from the `/emails` endpoint of <br /> the OAuth2 provider (obtained by appending `/emails` to the URL <br /> configured with `api_url`) | N/A                                                                                                                |
+
+## Configure a refresh token
```

Comment:
@IevaVasiljeva , should this be an H3 so that it is nested under "Configuration options"?

Reply:
It's not in the configuration option table, so I would keep it separate. I think it's fine to have another section with extra description for the refresh token step.

Reply:
Ok 👍